serial: Add driver for the LRW UART

GooTal opened the pull request:
fixed: #121 

注意代码仓库中 “riscv: defconfig: remove CONFIG_CMDLINE and CONFIG_CMDLINE_EXTEND as mainline does“ 的 commit message 中关联的 issue 号写错了，应该是 121，不是 151

    serial: Add driver for the LRW UART
    
    driver inclusion
    category: feature
    Link: https://github.com/RVCK-Project/rvck/issues/121
    
    --------------------------------
    
    This commit introduces a serial driver for the LRW UART controller,
    
    Key features implemented:
    - Support for FIFO mode (16-byte depth)
    - Baud rate configuration
    - Standard asynchronous communication formats:
      * Data bits: 5, 6, 7, 8, 9 bits
      * Parity: odd, even, fixed, none
      * Stop bits: 1 or 2 bits
    - Hardware flow control (RTS/CTS)
    - Multiple interrupt reporting mechanisms
    
    Signed-off-by: Wenhong Liu <liu.wenhong35@zte.com.cn>
    Signed-off-by: Qingtao Liu <liu.qingtao2@zte.com.cn>
    
@Kernalwh

Kernalwh committed:
dt-bindings: serial: Add binding for LRW UART

driver inclusion
category: feature
Link: https://github.com/RVCK-Project/rvck/issues/121

--------------------------------

Add documentation for LRW UART devicetree bindings.

Signed-off-by: Wenhong Liu <liu.wenhong35@zte.com.cn>
Signed-off-by: Qingtao Liu <liu.qingtao2@zte.com.cn>

Kernalwh committed:
serial: Add driver for the LRW UART

driver inclusion
category: feature
Link: https://github.com/RVCK-Project/rvck/issues/121

--------------------------------

This commit introduces a serial driver for the LRW UART controller and
add LRW UART in defconfig.

Key features implemented:
- Support for FIFO mode (16-byte depth)
- Baud rate configuration
- Standard asynchronous communication formats:
  * Data bits: 5, 6, 7, 8, 9 bits
  * Parity: odd, even, fixed, none
  * Stop bits: 1 or 2 bits
- Hardware flow control (RTS/CTS)
- Multiple interrupt reporting mechanisms

Signed-off-by: Wenhong Liu <liu.wenhong35@zte.com.cn>
Signed-off-by: Qingtao Liu <liu.qingtao2@zte.com.cn>

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/167/consoleFull

oervci commented:
开始测试

oervci commented:
Kernel build success!

oervci commented:
开始测试

oervci commented:
check patch done. log: https://jenkins.oerv.ac.cn/job/rvck-pipeline/job/check-patch/180/consoleFull

sterling-teng commented:
考虑到是原始补丁，并且有提交kernel.org社区，rvck这边可以先合并，等待upstream之后，记得backport上游版本。

进入物理机测试阶段。

GooTal commented:
@sterling-teng 
您好，想请教下如果提交内核模块，我们是要将Kconfig选项默认设为m么？

希望了解下欧拉的发行版，如何保证模块自动加载的？是有其他脚本写入自动加载的模块么（类似于安卓），我们是否需要添加什么内容保证系统自动加载模块呢？

sterling-teng commented:
> @sterling-teng 您好，想请教下如果提交内核模块，我们是要将Kconfig选项默认设为m么？
> 
> 希望了解下欧拉的发行版，如何保证模块自动加载的？是有其他脚本写入自动加载的模块么（类似于安卓），我们是否需要添加什么内容保证系统自动加载模块呢？

defconfig里设为M，总线探测到设备后，自动加载模块并绑定。

Kernalwh committed:
riscv: defconfig: remove CONFIG_CMDLINE and CONFIG_CMDLINE_EXTEND as mainline does

community inclusion
category: config
Link: https://github.com/RVCK-Project/rvck/issues/151

--------------------------------

Remove CONFIG_CMDLINE and CONFIG_CMDLINE_EXTEND as mainline does.

Signed-off-by: Wenhong Liu <liu.wenhong35@zte.com.cn>
Signed-off-by: Qingtao Liu <liu.qingtao2@zte.com.cn>

GooTal commented:
@sterling-teng 您好，已验证完成

sterling-teng commented on Documentation/devicetree/bindings/serial/lrw,lrw-uart.yaml:
```diff
@@ -0,0 +1,51 @@
+      current-speed = <115200>;
+    };
+
+...
```
结尾需要空一行，类似问题请自查自纠。
